compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -436,6 +438,43 @@ fn quoted_as_type(
     Ok(Value::Type(typ))
 }
 
+// fn as_integer(self) -> Option<(bool, u8)>
+fn type_as_integer(
+    mut arguments: Vec<(Value, Location)>,
+    return_type: Type,
+    location: Location,
+) -> IResult<Value> {
+    check_argument_count(1, &arguments, location)?;
+
+    let value = arguments.pop().unwrap().0;
+    let Value::Type(typ) = value else {
+        panic!("type_as_integer shold have been called with a type");
```

Comment:
I noticed we don't explicitly panic in this file, but instead return `InterpreterError`. Should these be changed to an error? (and if so, which error?)

Reply:
Yes, we should avoid panicking because this is still reachable if there was a type error.
We should issue a `TypeMismatch` error here. There's a few helper functions for this that other functions in this file use. E.g. `get_u32`, `get_quoted`, `get_trait_def`. We should add a `get_type` version following those examples.

Reply:
I tried doing this but I don't know what type to put into `expected`. Should I use just a type variable?

In any case, it should never panic because the type comes from the function meta.

Reply:
There's a type for types: `Type::Quoted(QuotedType::Type)` will be the `Type` type. In general most of the comptime-only types like Quoted, Expr, StructDefinition are variants of QuotedType.
`typ` here does not come from the function meta, it comes from unwrapping the value as a `Value::Type(_)` variant unless I'm misunderstanding.